### PR TITLE
Include lapack_wrap for Linux+LLVM and Windows builds (if lapack is found)

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* The `lapack_wrap` component of Draco has traditionally been omitted for all builds that didn't use Linux Makefiles and gcc. The component should work correctly for Linux+clang builds and for Windows builds.

### Purpose of Pull Request

* [Related to Redmine Issue #1218](https://rtt.lanl.gov/redmine/issues/1218)
* [Related to Redmine Issue #1219](https://rtt.lanl.gov/redmine/issues/1219)

### Description of changes

+ Enable lapack_wrap package for more build environments
  - allow if compiler is clang.
  - allow if generator is other than Makefiles.
+ Ensure lapack_wrap tests link to both lapack and blas libraries. This change
  was required on Windows.
+ Update, correct comments.
+ Allow CAFS to install the project.
  + Pass `CMAKE_INSTALL_PREFIX` to the Fortran sub-project.
  + Remove `NO_EXTERNAL_INSTALL` option requirement.
  + Allow the Fortran sub-project to build in parallel (`-j`).  
  + Teach the Fortran sub-project to run the `install` build target if `NO_EXTERNAL_INSTALL` is absent.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [ ] Travis CI checks pass
  * [ ] Code coverage does not decrease
  * [ ] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: config/CMakeAddFortranSubdirectory.cmake [hunk 1]
```diff
@@ -167,9 +180,12 @@ function(cmake_add_fortran_subdirectory subdir)
   set(multiValueArgs LIBRARIES TARGET_NAMES LINK_LIBRARIES DEPENDS CMAKE_COMMAND_LINE)
   cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
   if(NOT ARGS_NO_EXTERNAL_INSTALL)
-    message(FATAL_ERROR
-      "Option NO_EXTERNAL_INSTALL is required (for forward compatibility) "
-      "but was not given." )
+    message("
+-- The external_project ${ARGS_PROJECT} will be installed to to the location
```
Comment (RyanWollaeger) [inline config/CMakeAddFortranSubdirectory.cmake]: There is a very minor typo: "... installed to to ..." -> "... installed to ..."

File: config/CMakeAddFortranSubdirectory.cmake [hunk 2]
```diff
@@ -130,13 +130,26 @@ with the gfortran option." )
...
+"# This file generated by CMakeAddFortranSubdirectory.cmake
+#------------------------------------------------------------------------
+set(ENV{PATH} \"${CAFS_Fortran_COMPILER_PATH}\;\$ENV{PATH}\")
 set(VERBOSE ${ARGS_VERBOSE})
 if( VERBOSE )
-    message(\"${CMAKE_COMMAND} --build .\")
+    message(\"${CMAKE_COMMAND} ${build_command_args}\")
 endif()
-execute_process( COMMAND \"${CMAKE_COMMAND}\" --build . )")
+execute_process( COMMAND \"${CMAKE_COMMAND}\" ${build_command_args} )
```
Comment (RyanWollaeger) [inline config/CMakeAddFortranSubdirectory.cmake]: Do the build command args not need `--config` options (e.g. `--config Debug`) here?

File: config/CMakeAddFortranSubdirectory.cmake (hunk 1 above)
Comment (KineticTheory) [reply config/CMakeAddFortranSubdirectory.cmake]: I'll fix this.

File: config/CMakeAddFortranSubdirectory.cmake (hunk 2 above)
Comment (KineticTheory) [reply config/CMakeAddFortranSubdirectory.cmake]: `${build_command_args}` will either be
* `--config Debug -j [N]`, or
* `--config Debug -j [N] --target install`

**Pull request merged**